### PR TITLE
Improved comment parsing inside cleanup_comment()

### PR DESCRIPTION
**Pull request by papadokolos**
As discussed in #292, the method `cleanup_comment()` has been rewritten from scratch to improve its flexability and readability:
1. Since it's supposed to remove the brief comment, it is now getting the brief comment as a parameter (updated usage)
2. Improved handling of empty lines
3. `@param` tags are no longer supressed in the parsing

### CODE REVIEW

**Comment by niosus** [comment]
Thanks a lot for this PR. It looks good at the first glance. I will play around with it on the weekend probably and will merge it then. Thanks a lot! :+1:

**Comment by niosus** [comment]
Ok, after some testing, it crashes on me.

Given the function 
```c++
/// @brief What?
/// Some more text goes here.
void f(int a) {
  // cool code
}
```
Hovering on it gives me an error:
```
exception calling callback for <Future at 0x7f987013d510 state=finished raised ValueError>
Traceback (most recent call last):
  File "./python3.3/concurrent/futures/_base.py", line 296, in _invoke_callbacks
  File "/home/igor/.config/sublime-text-3/Packages/EasyClangComplete/EasyClangComplete.py", line 326, in info_finished
    (tooltip_request, result) = future.result()
  File "./python3.3/concurrent/futures/_base.py", line 394, in result
  File "./python3.3/concurrent/futures/_base.py", line 353, in __get_result
  File "./python3.3/concurrent/futures/thread.py", line 54, in run
  File "/home/igor/.config/sublime-text-3/Packages/EasyClangComplete/plugin/completion/lib_complete.py", line 273, in info
    cursor.referenced, self.cindex)
  File "/home/igor/.config/sublime-text-3/Packages/EasyClangComplete/plugin/clang/utils.py", line 349, in build_info_details
    cursor.raw_comment, cursor.brief_comment)
  File "/home/igor/.config/sublime-text-3/Packages/EasyClangComplete/plugin/clang/utils.py", line 450, in cleanup_comment
    clean_lines, brief_comment_to_remove)
  File "/home/igor/.config/sublime-text-3/Packages/EasyClangComplete/plugin/clang/utils.py", line 432, in find_brief_comment
    raise ValueError("comment_lines does contain a brief comment")
ValueError: comment_lines does contain a brief comment
```

File: plugin/clang/utils.py [hunk 1]
```diff
@@ -420,34 +423,39 @@ def build_objc_message_info_details(cursor):
...
-                non_brief_found = True
-                is_brief_comment = True
-                continue
-            non_brief_found = True
-            clean_lines.append(clean)
+        def find_brief_comment(comment_lines, brief_comment_to_remove):
+            for idx, line in enumerate(comment_lines):
+                if line.endswith(brief_comment_to_remove):
+                    return idx
+            raise ValueError("comment_lines does contain a brief comment")
```
Comment (niosus) [inline plugin/clang/utils.py]: This throws error. Details in https://github.com/niosus/EasyClangComplete/pull/293#issuecomment-318864066

Comment (papadokolos) [inline plugin/clang/utils.py]: While trying to solve this problem, I discovered that `cursor.brief_comment` can contain multiple lines from the raw comment joined into one line. The current logic assumes that a brief comment can not span multiple lines, thus causing the exception.

I'll have to rethink the way im locating and removing the brief comment within the raw comment. Another option is to let aside for the separation between brief and raw, and just show the full comment altogether. What do you think?

Comment (niosus) [inline plugin/clang/utils.py]: So the ugliness that you saw before in my code was caused by exactly this :) 

I definitely don't want the brief comment to be shown twice. Just showing the full comment might be a valid solution, but I still would prefer to parse it. 

I will probably have a look today at how I did it before and will try to figure out why the `@param` options are missing. That must be smth stupid :)

**Comment by niosus** [comment]
Hey @papadokolos I have used some of your ideas and have just simplified the things I have written before. Thanks for the inspiration! I will close this PR in favor of #298

**Pull request closed without merging**